compiler-core/src/javascript.rs

```diff
@@ -345,18 +345,22 @@ impl<'a> Generator<'a> {
         imports: &mut Imports<'a>,
         package: &'a str,
         module: &'a str,
-        as_name: &'a Option<ImportName>,
+        as_name: &'a Option<(AssignName, SrcSpan)>,
         unqualified: &'a [UnqualifiedImport],
     ) {
-        let module_name = as_name
-            .as_ref()
-            .map(|n| SmolStr::as_str(&n.name))
-            .unwrap_or_else(|| {
-                module
-                    .split('/')
-                    .last()
-                    .expect("JavaScript generator could not identify imported module name.")
-            });
+        let get_name = |module: &'a str| {
+            module
+                .split('/')
+                .last()
+                .expect("JavaScript generator could not identify imported module name.")
+        };
+
+        let (discarded, module_name) = match as_name {
+            None => (false, get_name(module)),
+            Some((AssignName::Discard(_), _)) => (true, get_name(module)),
+            Some((AssignName::Variable(name), _)) => (false, name.as_str()),
```

Comment:
This isn't right, it's using the full name of the module but that may not be a valid JavaScript variable.

It'll also need to handle that two modules could have the same last segment as their name, but now one of them could have been discarded.

Reply:
If I understand the first objection correctly, it's actually not using the full name(do you mean `a/b/c`?). Because I'm using the `get_name` lambda to get the last part. Is that the problem or am I misunderstanding it?

Discarded modules are actually handled on line 379. If the module is discarded, an empty vector is passed to `imports.register_module` which then proceeds to only register unqualified imports.

Reply:
Sorry, I misread this line and didn't see the `get_name`. Thank you